gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -192,21 +192,19 @@ def self._check_final_ancestors(target, target_ancestors, source_method_names, s
   end
 
   def self.add_module_with_final_method(mod, method_name, is_singleton_method)
-    m = is_singleton_method ? mod.singleton_class : mod
-    mid = m.object_id
-    methods = @modules_with_final[mid]
+    methods = @modules_with_final[m]
```

Comment:
Looks like you need to put back the definition of `m` here.

Reply:
Good catch. This was a leak from another change I was making (foreshadowing! :eyes)